runtime-sdk/modules/evm/src/state.rs

```diff
@@ -70,26 +71,27 @@ where
 
     // These values are used to derive the confidential store nonce:
     let round = ctx.runtime_header().round;
-    let instance_count: usize = {
-        // One Context is used per tx batch, so the instance count will monotonically increase.
-        let cnt = *ctx
-            .value(CONTEXT_KEY_CONFIDENTIAL_STORE_INSTANCE_COUNT)
+    let instance_count: usize = CurrentState::with(|state| {
+        // One state is used per tx batch, so the instance count will monotonically increase.
```

Comment:
Shooting from the hip: You mention in the PR description that `State` "supports nesting and transactions". Is this comment still right? There's one State that keeps track of forks and reverts, rather than a tree of States?

Reply:
Values behave differently as each block-wide value needs to exist for the duration of the whole block and implements its own semantics.

So block-wide values are not affected by reverts and there is also no clear need for this. We should probably update the documentation on value accessors to make this clear.

Reply:
Updated documentation for `block_value` and `local_value` methods.